### PR TITLE
Pod log location must validate container if provided

### PR DESCRIPTION
**Pull request by fabianofranz**
When checking the log URL for a pod container we must validate the container name if provided, otherwise the method will return an URL that is actually not valid and results in an error when called, e.g. `the server could not find the requested resource ( Pod mypod)`.

### CODE REVIEW

File: pkg/registry/pod/strategy.go [hunk 1]
```diff
@@ -240,6 +240,7 @@ func LogLocation(getter ResourceGetter, connInfo client.ConnectionInfoGetter, ct
 	}
 
 	// Try to figure out a container
+	// If a container was provided, it must be valid
 	container := opts.Container
 	if container == "" {
```
Comment (kargakis) [inline pkg/registry/pod/strategy.go]: Mind changing this to `len(container) == 0`

File: pkg/registry/pod/strategy.go [hunk 2]
```diff
@@ -247,6 +248,17 @@ func LogLocation(getter ResourceGetter, connInfo client.ConnectionInfoGetter, ct
 		} else {
 			return nil, nil, errors.NewBadRequest(fmt.Sprintf("a container name must be specified for pod %s", name))
 		}
+	} else {
+		found := false
```
Comment (kargakis) [inline pkg/registry/pod/strategy.go]: A helper function for this?

**Comment by liggitt** [comment]
Lgtm, update with feedback from @kargakis?

**Comment by liggitt** [comment]
Build error
```
Error pulling image (0.3) from gcr.io/google_containers/kube-registry-proxy, Get https://storage.googleapis.com/artifacts.google-containers.appspot.com/containers/images/9b9342c134bbda6f57c95c52cafd4ae40333dd6cc0fce4bace9296f957042d22/ancestry: dial tcp 74.125.202.128:443: i/o timeout
!!! Error in /jenkins-master-data/jobs/kubernetes-pull-build-test-e2e-gce/workspace/hack/e2e-internal/../../cluster/gce/../../cluster/../build/../build/common.sh:817
  'docker pull "${addon_path}"' exited with status 1
Call stack:
  1: /jenkins-master-data/jobs/kubernetes-pull-build-test-e2e-gce/workspace/hack/e2e-internal/../../cluster/gce/../../cluster/../build/../build/common.sh:817 kube::release::write_addon_docker_images_for_server(...)
  2: /jenkins-master-data/jobs/kubernetes-pull-build-test-e2e-gce/workspace/hack/e2e-internal/../../cluster/gce/../../cluster/../build/../build/common.sh:719 kube::release::package_server_tarballs(...)
  3: /jenkins-master-data/jobs/kubernetes-pull-build-test-e2e-gce/workspace/hack/e2e-internal/../../cluster/gce/../../cluster/../build/../build/common.sh:654 kube::release::package_tarballs(...)
  4: /jenkins-master-data/jobs/kubernetes-pull-build-test-e2e-gce/workspace/hack/e2e-internal/../../cluster/gce/../../cluster/../build/release.sh:40 main(...)
Exiting with status 1
!!! [1129 05:56:17] unable to pull or write addon image
!!! Error in /jenkins-master-data/jobs/kubernetes-pull-build-test-e2e-gce/workspace/hack/e2e-internal/../../cluster/gce/../../cluster/../build/../build/common.sh:808

```

@k8s-bot test this

File: pkg/registry/pod/strategy.go (hunk 1 above)
Comment (fabianofranz) [inline pkg/registry/pod/strategy.go]: Done.

File: pkg/registry/pod/strategy.go (hunk 2 above)
Comment (fabianofranz) [inline pkg/registry/pod/strategy.go]: Done.

**Comment by fabianofranz** [comment]
Updated addressing @kargakis comments, @liggitt needs lgtm again.

**Comment by wojtek-t** [comment]
@k8s-bot e2e test this please

**Pull request merged**